### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by cleydyr**
Hey, @mbaeuerle .

I've introduced automatic source formatting and [Java] automated tests to the project. I've also performed some refactorings in the first commits that will be useful in the future to make the project more testable.

Can you have a look at it?

Thanks

### CODE REVIEW

File: src/main/java/at/laborg/briss/BrissSwingGUI.java [hunk 1]
```diff
@@ -508,19 +548,18 @@ public void setDefinedSizeSelRects() {
...
         }
 
         // get user input
         // maximums are used as a default
         String input = JOptionPane.showInputDialog(Messages.getString("BrissGUI.size"), defInput);
 
-        if (input == null || input.equals("")) //$NON-NLS-1$
-            return;
+        if (input == null || input.equals("")) // $NON-NLS-1$
+        return;
```
Comment (mbaeuerle) [inline src/main/java/at/laborg/briss/BrissSwingGUI.java:559]: This formatting seems to be wrong?

File: src/main/java/at/laborg/briss/BrissSwingGUI.java [hunk 2]
```diff
@@ -563,19 +603,18 @@ public void setPositionSelRects() {
...
         }
 
         // get user input
         // minimums are used as a default
         String input = JOptionPane.showInputDialog(Messages.getString("BrissGUI.position"), defInput);
 
-        if (input == null || input.equals("")) //$NON-NLS-1$
-            return;
+        if (input == null || input.equals("")) // $NON-NLS-1$
+        return;
```
Comment (mbaeuerle) [inline src/main/java/at/laborg/briss/BrissSwingGUI.java:614]: Same as before, maybe we should add the parenthesis so the formatter can also understand it better

File: src/main/java/at/laborg/briss/cli/BrissCMD.java [hunk 3]
```diff
@@ -74,23 +71,26 @@ public static void autoCrop(final String[] args) {
         System.out.println("Calculating crop rectangles.");
         try {
             for (PageCluster cluster : clusterDefinition.getClusterList()) {
-                float[] auto = CropFinder.getAutoCropFloats(cluster.getImageData().getPreviewImage());
+                float[] auto =
+                        CropFinder.getAutoCropFloats(cluster.getImageData().getPreviewImage());
```
Comment (mbaeuerle) [inline src/main/java/at/laborg/briss/cli/BrissCMD.java:75]: Is there a way to set the max width for the formatter? I would like to set it to 120!
In this case the line only has 103 characters, no IMO no need to wrap it already as screens are rather wide these days 😃

File: src/main/java/at/laborg/briss/gui/MergedPanel.java [hunk 4]
```diff
@@ -276,8 +280,7 @@ public void setSelCropSize(int width, int height) {
     public void resizeSelCrop(int width, int height) {
         for (DrawableCropRect crop : crops) {
             if (crop.isSelected()) {
-                if (((width < 0) && (crop.width <= -width)) || ((height < 0) && (crop.height <= -height)))
-                    return;
+                if (((width < 0) && (crop.width <= -width)) || ((height < 0) && (crop.height <= -height))) return;
```
Comment (mbaeuerle) [inline src/main/java/at/laborg/briss/gui/MergedPanel.java:283]: I'm not  a big fan of putting the one line after the `if`. But I'm also not a fan of omitting the parenthesis 😄.
So maybe we should add the parenthesis one day (no need to do that in this PR) and this issue gets resolved anyway.

**Review by mbaeuerle — COMMENTED** [review]
Hi @cleydyr! Thanks for your contribution!
I like the addition of automated tests and formatting!
I put some remarks directly into comments. Also would love to see this shortened to e.g. 3 commits:
* Refactorings
* Tests
* Initial reformatting

File: build.gradle [hunk 5]
```diff
@@ -2,6 +2,8 @@ plugins {
     id 'org.openjfx.javafxplugin' version '0.0.8'
     id 'java'
     id 'application'
+    id 'eclipse'
```
[the remark this answers is not in the document]
Comment (cleydyr) [reply build.gradle:5]: I normally use Eclipse for editing code and this plugins eases the set up of the project for that IDE.

File: src/main/java/at/laborg/briss/BrissSwingGUI.java (hunk 1 above)
Comment (cleydyr) [reply src/main/java/at/laborg/briss/BrissSwingGUI.java:559]: That's weird. I'll make sure to double check it and use curly braces.

File: src/main/java/at/laborg/briss/cli/BrissCMD.java (hunk 3 above)
Comment (cleydyr) [reply src/main/java/at/laborg/briss/cli/BrissCMD.java:75]: I think there is. I'll check that out and set to 120. But the issue is not about the screen width but the amount of movement the reader' eyes will have to make in order to read the entire line.

File: src/main/java/at/laborg/briss/gui/MergedPanel.java (hunk 4 above)
Comment (cleydyr) [reply src/main/java/at/laborg/briss/gui/MergedPanel.java:283]: You mean the curly braces around the if? Totally agreed. :D I'll see if I can include this as a rule for the formatter.

**Comment by cleydyr** [comment]
Let's merge this one (or an equivalent) first so that we can add a test for #53 . What do you think, @mbaeuerle ?

File: src/main/java/at/laborg/briss/cli/BrissCMD.java (hunk 3 above)
Comment (cleydyr) [reply src/main/java/at/laborg/briss/cli/BrissCMD.java:75]: No luck. With Google/Palantir the max line length can't be changed. :(

**Comment by cleydyr** [comment]
Closing in favor of https://github.com/mbaeuerle/Briss-2.0/pull/54

**Pull request closed without merging**